sklearn/svm/_base.py

```diff
@@ -266,6 +266,15 @@ def fit(self, X, y, sample_weight=None):
             self.intercept_ *= -1
             self.dual_coef_ = -self.dual_coef_
 
+        # If the number of models optimized by libSVM is one, get the number of
+        # iterations as an integer instead of ndarray.
+        if (
+            self._impl in ["c_svc", "nu_svc"] and len(self.classes_) <= 2
+        ) or self._impl in ["one_class", "epsilon_svr", "nu_svr"]:
```

Comment:
If this is just to check whether the number of models is 1, why not test for `len(self._num_iter)` instead then? Unless it's the case that sometimes `len(self._num_iter)` is 1, but we still want to expose an ndarray.

Reply:
You are right. I was trying to be super explicit about this, so we know when (which combination of SVM type and the number of classes) we will have an int and when we will have an array. But I think your suggestion leaves a cleaner code.